### PR TITLE
Update documentation/Makefile

### PR DESCRIPTION
**Pull request by yangyansong-adbe**
- Update the main README and the sample app's README
- Add two make commands (`version` and `archive`) for releasing Roku SDK

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,35 @@
...
+clean:
+	(rm -rf $(OUTPUT_DIR))
+	
+packing:clean
+	@echo "######################################################################"
+	@echo "##### packing"
+	@echo "######################################################################"
+	@echo git-hash=$(GIT_HASH)
+
+	mkdir -p $(OUTPUT_DIR)
```
Comment (addb) [inline Makefile:25]: As discussed we could generate the zip with the version number. And we include sample app with the final build of the SDK.

Sample app changes script can be added after the alpha release.

File: sample/simple-videoplayer-channel/README.md [hunk 2]
```diff
@@ -1 +1,21 @@
...
+
+To run the project on your Roku device, follow the steps below to set it up beforehand.
+
+## Install the Roku SDK
+
+There are two options for installing the Roku SDK in this sample project:
+
+- Copy the SDK files to this sample project by following the [install instruction](../../Documentation/getting-started.md).
+
+- If you downloaded the entire GitHub repository, link the SDK source files to this project by running `make link-sdk` in the current directory.
```
Comment (addb) [inline sample/simple-videoplayer-channel/README.md:13]: Can we avoid this by attaching and not linking the latest code to the sample apps? Same as we discussed for sample app with our release candidate.

File: Makefile [hunk 3]
```diff
@@ -1,3 +1,35 @@
...
+	touch $(OUTPUT_DIR)/$(INFO_TXT_FILE)
+	@echo git-hash=$(GIT_HASH) >> $(OUTPUT_DIR)/$(INFO_TXT_FILE)
+	@echo version=$(VERSION) >> $(OUTPUT_DIR)/$(INFO_TXT_FILE)
+	cp -r $(SDK_COMPONENTS_DIR) $(OUTPUT_DIR)
+	cp $(SDK_MAIN_FILE) $(OUTPUT_DIR)/
+
+	 zip -r ./out/AdobeEdge-$(VERSION).zip $(OUTPUT_DIR)/*
```
Comment (yangyansong-adbe) [inline Makefile:32]: @addb I just remembered that the version string was included in the zip.

File: Makefile (hunk 1 above)
Comment (yangyansong-adbe) [reply Makefile:25]: @addb  https://github.com/adobe/aepsdk-roku/issues/25 is created to track the task.

**Review by addb — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -1,3 +1,35 @@
...
-	(npm install brighterscript -g)
\ No newline at end of file
+	(npm install brighterscript -g)
+
+clean:
+	(rm -rf $(OUTPUT_DIR))
+	
+archive:clean
+	@echo "######################################################################"
+	@echo "##### packing"
```
Comment (addb) [inline Makefile:21]: @echo "##### Archiving AdobeEdge SDK"

File: Documentation/api-reference.md [hunk 5]
```diff
@@ -115,24 +119,27 @@ sendEdgeEvent: function(xdmData as object, callback = _adb_default_callback as f
 - `@param [optional] context as dynamic : context to be passed to the callback function`
 
 > This function will automatically add an identity property, the Experience Cloud Identifier (ECID), to each Edge network request within the Experience event's `XDM IdentityMap`.
+> Also `ImplementationDetails` are automatically collected and are sent with every Experience Event. If you would like to include this information in your dataset, add the "Implementation Details" field group to the schema tied to your dataset.
```
Comment (addb) [inline Documentation/api-reference.md:122]: Do we need to provide this context with the API description?

File: Documentation/api-reference.md [hunk 6]
```diff
@@ -144,13 +151,31 @@ sendEdgeEvent: function(xdmData as object, callback = _adb_default_callback as f
...
+##### Syntax
+
+```brightscript
+resetIdentities: function() as void
+```
+
+##### Example
+
+```brightscript
+  m.adobeEdgeSdk.resetIdentities()
```
Comment (addb) [inline Documentation/api-reference.md:167]: nit: indentation

File: Documentation/api-reference.md (hunk 5 above)
Comment (yangyansong-adbe) [reply Documentation/api-reference.md:122]: Sounds good!

**Pull request merged**